tests/wallets/ethereum.spec.ts

```diff
@@ -130,10 +139,67 @@ describe("Ethereum Wallet", () => {
       const wallet = await Web3EthereumWallet.newInstance(web3, privateKey);
 
       const amountFounded = await fundAddressOfPrivateKey(web3, privateKey);
-      const balance = await wallet.getNominalBalance();
+      const balance = await wallet.getBalance(Asset.ether);
       const balanceAfter = web3.utils.toWei(balance.toFixed(10));
       expect(balanceAfter.toString()).toEqual(amountFounded.toString());
     }),
     20000
   );
+
+  test(
+    "given deployed ERC20 token contract and initial minting, returns the balance",
+    containerTest(parityTestContainer, async ({ container }) => {
+      const web3 = new Web3(
+        new Web3.providers.HttpProvider(
+          `http://localhost:${container.getMappedPort(8545)}`
+        )
+      );
+      const wallet = await Web3EthereumWallet.newInstance(web3, privateKey);
+      const address = wallet.getAddress();
+
+      await fundAddressOfPrivateKey(web3, privateKey);
+
+      // Deploy token contract
+      const deploymentReceipt = await wallet.deployContract({
+        gasPrice: new BN(0),
+        gasLimit: new BN(4_000_000),
+        data: tokenContractCode
+      });
+      const contractAddress = deploymentReceipt.contractAddress;
+      expect(contractAddress).not.toBeUndefined();
+
+      // wait for parity to mine the transaction
+      // parity sometimes takes up to 4 seconds to actually mine the transaction even though it already returned the receipt. this is how we cater for that.
+      // See https://github.com/paritytech/parity-ethereum/issues/10672 for details
+      await sleep(5000);
```

Comment:
🔧 What if it takes longer than 5 seconds? 
It might be better to actively query for the transaction to be confirmed every X seconds. This might also be faster in the best case.

Reply:
It's a copy paste but I'll look into it.